internal/oci/runtime_oci.go

```diff
@@ -871,15 +871,14 @@ func (r *runtimeOCI) StopLoopForContainer(c *Container) {
 			log.Warnf(ctx, "Stopping container %v with stop signal timed out. Killing", c.ID())
 			if _, err := r.runtimeCmd("kill", c.ID(), "KILL"); err != nil {
 				log.Errorf(ctx, "Killing container %v failed: %v", c.ID(), err)
+				targetTime = time.Now().Add(10 * time.Millisecond)
```

Comment:
Doesn't this still mean that in case of a container that can't be killed with -9 due to uninetteruptible sleep, we'd be logging every 10 ms?

Reply:
@andreaskaris, assuming that we've attempted to send the graceful termination signal (SIGTERM) already and then _at least once_ send the SIGKILL signal, then the aim with the new changes is to add this 10 ms delay/stagger and also bail out for any process which is in the state such as "D", "X" or "Z". Otherwise, yes, it would also be a lot of log entries (10 ms is not a lot relatively).

As for the "D" state, to us, it does not matter whether it's killable or uninterruptible—at this point, we are already on the path to termination. There should be at least one pending signal (since we should have attempted to send SIGKILL at least once) awaiting such a process to awake (when it will be delivered). There is also no easy way to see whether the "D" status has a TASK_KILLABLE or TASK_UNINTERRUPTIBLE flag set (userspace seems to have limited visibility via /proc in that specific case).